rotkehlchen/chain/zksync_lite/manager.py

```diff
@@ -494,34 +494,37 @@ def get_db_transactions(
         """Gets any zksynclite transactions from the DB depending on the given filter"""
         transactions = []
         with self.database.conn.read_ctx() as cursor:
-            cursor.execute(
+            zksync_lite_data = cursor.execute(
                 f'SELECT identifier, tx_hash, type, timestamp, block_number, from_address, '
                 f'to_address, asset, amount, fee FROM zksynclite_transactions{queryfilter}',
                 bindings,
+            ).fetchall()
```

Comment:
You can avoid executing this `fetchall` if you do this query after saving `swap_data = {row[0]: row[1:] for row in cursor}`.

Fetchall is bad because it is a for loop that consumes all the entries into memory so you process twice the list